theseus/core/variable.py

```diff
@@ -121,3 +121,27 @@ def as_variable(
     if isinstance(value, float):
         tensor = tensor.view(1, 1)
     return Variable(tensor, name=name)
+
+
+# This context can be used to temporarily use a masked version of a list of
+# variables. For example,
+#
+# with masked_variables([v1, v2], mask):
+#   some_computation(v1, v2)
+#
+# Performs `some_computation()` using v1.tensor[mask] and v2.tensor[mask].
```

Comment:
In practice, we would use this to mask any dimension (not necessarily just batch indices), correct?